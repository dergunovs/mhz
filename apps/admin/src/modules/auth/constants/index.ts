export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
