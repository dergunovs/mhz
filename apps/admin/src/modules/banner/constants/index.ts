export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
