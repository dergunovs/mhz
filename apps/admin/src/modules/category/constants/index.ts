export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
