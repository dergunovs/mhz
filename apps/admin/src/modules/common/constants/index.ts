export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
