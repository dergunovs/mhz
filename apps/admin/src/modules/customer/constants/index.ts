export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
