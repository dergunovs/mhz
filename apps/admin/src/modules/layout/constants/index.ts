import { INavItem } from '@/layout/interface';

export const NAV_ITEMS: INavItem[] = [
  { _id: 0, url: '/', title: 'Главная' },
  { _id: 1, url: '/users', title: 'Пользователи' },
  { _id: 2, url: '/shipments', title: 'Доставка' },
  { _id: 3, url: '/orders', title: 'Заказы' },
  { _id: 4, url: '/categories', title: 'Категории' },
  { _id: 5, url: '/products', title: 'Товары' },
  { _id: 5, url: '/manufacturers', title: 'Производители' },
];
