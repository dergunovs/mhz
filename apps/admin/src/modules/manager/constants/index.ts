export const URL_MANAGER = '/managers';
