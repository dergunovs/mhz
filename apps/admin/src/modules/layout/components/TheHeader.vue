<template>
  <header :class="$style.header">
    <div>Лого</div>
    <Button @click="logout">Выйти</Button>
  </header>
</template>

<script setup lang="ts">
import { Button } from 'ant-design-vue';

import { logout } from '@/auth/composables';
</script>

<style module lang="scss">
.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 16px;
  background-color: var(--color-primary-light);
}
</style>
