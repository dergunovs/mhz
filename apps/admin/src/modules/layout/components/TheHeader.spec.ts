import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import * as helpers from 'mhz-helpers';
import { dataTest } from 'mhz-helpers';

import TheHeader from './TheHeader.vue';

import { wrapperFactory } from '@/common/test';
import { TOKEN_NAME, URL_LOGIN } from '@/auth/constants';

const spyLogout = vi.spyOn(helpers, 'logout');
const deleteAuthHeader = vi.spyOn(helpers, 'deleteAuthHeader');

const headerLogout = dataTest('header-logout');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(TheHeader, {});
});

enableAutoUnmount(afterEach);

describe('TheHeader', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(TheHeader)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('handles logout by button click', async () => {
    expect(spyLogout).toBeCalledTimes(0);

    await wrapper.findComponent(headerLogout).trigger('click');

    expect(spyLogout).toBeCalledTimes(1);
    expect(spyLogout).toBeCalledWith(URL_LOGIN, deleteAuthHeader, TOKEN_NAME);
  });
});
