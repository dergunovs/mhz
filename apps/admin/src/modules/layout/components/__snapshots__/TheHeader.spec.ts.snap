// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheHeader > matches snapshot 1`] = `
"<header class="_header_e46623">
  <div class="_logoAndSearch_e46623"><a to="/main" aria-label="Logo">
      <anonymous-stub class="_logo_e46623"></anonymous-stub>
    </a>
    <the-search-stub></the-search-stub>
  </div><button layout="plain" data-test="header-logout"> Logout </button>
</header>"
`;
